packages/package_info_plus/package_info_plus/lib/package_info_plus.dart

```diff
@@ -33,12 +33,54 @@ class PackageInfo {
 
   /// Retrieves package information from the platform.
   /// The result is cached.
-  static Future<PackageInfo> fromPlatform() async {
+  ///
+  /// The [baseUrl] parameter is for web use only and the other platforms will
+  /// ignore it.
+  ///
+  /// ## Web platform
+  ///
+  /// In a web environment, the package uses the `version.json` file that it is
+  /// generated in the build process.
+  ///
+  /// The package will try to locate this file in 3 ways:
+  ///
+  ///   * If you provide the optional custom [baseUrl] parameter, it will be
+  ///     used as the first option where to search. Example:
+  ///
+  ///     ```dart
+  ///     await PackageInfo.fromPlatform(baseUrl: 'https://cdn.domain.com/with/some/path/');
+  ///     ```
+  ///
+  ///     With this, the package will try to search the file in `https://cdn.domain.com/with/some/path/version.json`
+  ///
+  ///   * The second option where it will search is the [assetBase] parameter
+  ///     that you can pass to the Flutter Web Engine when you initialize it.
+  ///
+  ///     ```javascript
+  ///     _flutter.loader.loadEntrypoint({
+  ///         onEntrypointLoaded: async function(engineInitializer) {
+  ///           let appRunner = await engineInitializer.initializeEngine({
+  ///             assetBase: "https://cdn.domain.com/with/some/path/"
+  ///           });
+  ///           appRunner.runApp();
+  ///         }
+  ///     });
+  ///     ```
+  ///
+  ///     For more information about the Flutter Web Engine initialization see here:
+  ///     https://docs.flutter.dev/platform-integration/web/initialization#initializing-the-engine
+  ///
+  ///   * Finally, if none of the previous locations return the `version.json` file,
+  ///     the package will use the browser window base URL to resolve its location.
+  static Future<PackageInfo> fromPlatform({String? baseUrl}) async {
```

Comment:
Would it be possible to pass here the full `version.json` path instead of the `baseUrl`, or said differently, is there a reason why the `baseUrl` is preferred?

I know that we can reuse the `versionJsonUrl` and it also adds the "cache buster", but I am thinking of giving full flexibility, so users can provide the `Uri` directly as it will give them more control.

Reply:
Well, I feel like the right way is to allow only to indicate the base URL where the `version.json` is located. It is like less implementation details dependent and it is consistent with the other 2 ways the package gets the version file.

On the other hand, the `baseUrl` is giving you the same flexibility as the full `Uri`, excepting the case you want to rename the file, which it is something that I don't feel we should support. `version.json` is a private artifact as far as I know.

Reply:
Yes, I see both points. My idea on providing the version.json URI directly was mostly because we had complaints regarding the cache-busting mechanism, so I thought that with the full path they can decide if they want to apply any cache busting query params or not.

Anyway, let's go with your implementation and see how the feedback goes!

Reply:
Yes, I saw that issue https://github.com/fluttercommunity/plus_plugins/issues/2675

Anyway, the source of that problem I think is the nature of the web implementation, that is dependent of an external resource that needs to be loaded with a fetch request (GET version.json) and as all web resources, it has its own cache.

You can remove the buster cache, but then you will depend of the browser cache invalidation. Your `main.dart.js` and your `version.json` can be invalidated in different moments, so you will be in the same situation.

And this only thinking in the browser cache invalidation, but it could be other cache layers, like for example an AWS CloudFront cache invalidation layer, or others.

It is a complicated situation, to be honest.

We could add a new parameter to opt out the cache buster. Maybe that could mitigate the issue, but in my view, it will be there yet, just happening sometimes, instead always.

Or, we could implement a solution no dependent of the `version.json` file. For example, in the web build process of the app, the package could inject the same information that we can find in the `version.json` file in a dart file. In that way, you can know exactly which version of the app you are seeing in your browser, because the information it is in the same `main.dart.js` of your app.

Of course, this is just a theoretical solution, I don't know if it could be implemented even.

I will put this comment in the original issue, so we keep the discussion there if you agree.

Reply:
Yep, I am happy to explore alternatives to the `version.json` file. We can keep the discussion on that ticket.